charmcraft.yaml

```diff
@@ -46,3 +46,12 @@ actions:
     - model
     - prompt
     additionalProperties: false
+  pull:
+    description: "Downloads and serves the model you specify. Find available models here: https://ollama.com/library"
+    params:
+      model:
+        type: string
```

Comment:
There are pros and cons for having this of enum type.
If new model names are introduced infrequently then an enum could be a good choice.

Reply:
Since the models are hosted on a remote library (https://ollama.com/library), new model names are introduced frequently, my judgement is string is better for this one.